### PR TITLE
First Half of IPC Redesign

### PR DESCRIPTION
**Pull request by anthonyfabius**
# Pull Request

## Description

This PR introduces the initial functionality for the IPC redesign in paradise. It revamps how we create and handle the mailbox & its messages, includes more util functions across different files, and alters current static addresses of processes, mailbox, & the pcb.

The mailbox currently supports only 1 page of data (4KiB) as growing the mailbox will come in the next ipc pr. Blocking is also no longer supported but should be in the next pr. ParadiseDocs with the information regarding this design will be populated soon.

> [!NOTE]
> The new process start origin is 0x420000 meaning all paradise programs that enter at 0x402000 will no longer work with these changes.

## How Has This Been Tested?

I ran a series of custom reader and writer programs that scrambled the order of reads and writes to ensure the mailbox was working as expected. I didn't find any issues with the mailbox so far but my tests weren't extensive enough to say the edge cases are covered. I believe they are but before merging feel free to try and break it.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by anthonyfabius** [comment]
This isn't rebased on #32 yet btw

File: src/ipc/mailbox.c [hunk 1]
```diff
@@ -1,175 +1,172 @@
...
-    mailbox->tail = &mailbox->data;
-    mailbox->size = 0;
-    mailbox->capacity = sizeof(mailbox->data);
-    mailbox->next_page = mailbox;
-    mailbox->prev_page = mailbox;
-    return mailbox;
+static void bytes_to_message_header(
+    MailboxMessageHeader *message_header, const void *data
+) {
+    pmemcpy(message_header, data, sizeof(MailboxMessageHeader));
```
Comment (joyal-mathew) [inline src/ipc/mailbox.c]: This could be an assignment
```c
*message_header = *((const MailboxMessageHeader *) data);
```
Doesn't matter that much.

**Review by joyal-mathew — DISMISSED** [review]
𓆏

**Review by joyal-mathew — APPROVED** [review]
𓆏
